helix-core/src/search.rs

```diff
@@ -22,6 +39,38 @@ pub fn find_nth_next(text: RopeSlice, ch: char, mut pos: usize, n: usize) -> Opt
     Some(pos - 1)
 }
 
+/// Like find_nth_next, but stops at the first newline character.
+pub fn find_nth_next_until_newline<M: CharMatcher>(
+    text: RopeSlice,
+    char_matcher: M,
+    mut pos: usize,
+    n: usize,
+) -> Option<usize> {
+    if pos >= text.len_chars() || n == 0 {
+        return None;
+    }
+
+    let mut chars = text.chars_at(pos);
```

Comment:
Rather than a separate method that has to check for `\n` & `\r`, you could modify `find_nth_prev` to use a matcher/pattern and pass in a text slice of a single line: `text.line(n)`

Reply:
Done.
This made the function `find_next_char_until_newline` a bit more complicated, though.
Any idea if I could have done it in a simpler way?